cosmovisor/cmd/cosmovisor/cmd/help.go

```diff
@@ -0,0 +1,101 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/cosmos/cosmos-sdk/cosmovisor"
+	"os"
+)
+
+// ShouldGiveHelp checks the env and provided args to see if help is needed or being requested.
+// Help is needed if the os.Getenv(EnvName) env var isn't set.
+// Help is requested if the first arg is "help" or "--help"; or the only arg is "-h"
+func ShouldGiveHelp(args []string) bool {
+	if len(os.Getenv(cosmovisor.EnvName)) == 0 {
+		return true
+	}
+	if len(args) == 0 {
+		return false
+	}
+	return args[0] == "help" || args[0] == "--help" || (len(args) == 1 && args[0] == "-h")
+}
+
+// DoHelp outputs help text, config info, and attempts to run the binary with the --help flag.
+func DoHelp() {
+	// Output the help text
+	fmt.Println(GetHelpText())
+	// If the config isn't valid, say what's wrong and we're done.
+	cfg, cerr := cosmovisor.GetConfigFromEnv()
+	switch err := cerr.(type) {
+	case nil:
+		// Nothing to do. Move on.
+	case *cosmovisor.MultiError:
+		fmt.Fprintf(os.Stderr, "[cosmovisor] multiple configuration errors found:\n")
+		for i, e := range err.GetErrors() {
+			fmt.Fprintf(os.Stderr, "  %d: %v\n", i+1, e)
+		}
+		return
+	default:
+		fmt.Fprintf(os.Stderr, "[cosmovisor] %v\n", err)
+		return
+	}
+	// If the config's legit, output what we see it as.
+	fmt.Println("[cosmovisor] config is valid:")
+	fmt.Println(cfg.DetailString())
+	// Attempt to run the configured binary with the --help flag.
+	if err := cosmovisor.RunHelp(cfg, os.Stdout, os.Stderr); err != nil {
+		fmt.Fprintf(os.Stderr, "[cosmovisor] %v\n", err)
+	}
+}
+
+// GetHelpText creates the help text multi-line string.
+func GetHelpText() string {
+	return fmt.Sprintf(`Cosmosvisor - A process manager for Cosmos SDK application binaries.
+
+Cosmovisor monitors the governance module for incoming chain upgrade proposals.
+If it sees a proposal that gets approved, cosmovisor can automatically download
+the new binary, stop the current binary, switch from the old binary to the new one,
+and finally restart the node with the new binary.
```

Comment:
Cosmovisor is a wrapper for a Cosmos SDK based App (set using the required
`DAEMON_NAME` env variable). It starts the App by passing all positional
arguments and monitors the `$DAEMON_HOME/data/upgrade-info.json` file to
perform an update. The `upgrade-info.json` is created by the App `x/upgrade`
module when the blockchain height reaches an approved upgrade proposal. The
file includes data from the proposal. Cosmovisor tries to interpret that data
to perform an update: switch a current binary and restarts the App.

Reply:
FYI: cosmovisor doesn't observe `x/gov`. It doesn't stop the App nigher - the App stops by it's own.

Reply:
I had just copied this from the top of the `cosmovisor` readme. I split the first sentence, putting the first part as a short description on the first line, then put the rest as the extra info here.

I've updated it to the text you have in the first comment, though.